localstack-core/localstack/services/apigateway/next_gen/execute_api/gateway_response.py

```diff
@@ -1,125 +1,143 @@
-from localstack.aws.api.apigateway import GatewayResponseType
+from functools import cache
 
+from localstack.aws.api.apigateway import (
+    GatewayResponse,
+    GatewayResponseCode,
+    GatewayResponseType,
+)
 
-class BaseGatewayResponse(Exception):
+DEFAULT_GATEWAY_RESPONSE = {
+    "responseParameters": {},
+    "responseTemplates": {"application/json": '{"message":$context.error.messageString}'},
+}
+
+
+@cache
+def build_default_response(type: GatewayResponseType) -> dict:
+    response = GatewayResponse(**DEFAULT_GATEWAY_RESPONSE, responseType=type, defaultResponse=True)
+    if status_code := GatewayResponseCode[type]:
+        response["statusCode"] = status_code
+    return response
+
+
+class BaseGatewayException(Exception):
     """
     Base class for all Gateway exceptions
     Do not use this class directly. Instead, subclass from Default4xxError or Default5xxError.
     """
 
     message: str = "Unimplemented Response"
-    status_code: int = 500
+    status_code: str = ""
```

Comment:
Setting them as string means we are always and only dealing with `str` for the status code as we can pass a `str` to `werkzeug.Response` and we don't need to change it's type ever. Otherwise we need to change it's type when building the default in my utility.